ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManagerMetrics.java

```diff
@@ -223,13 +224,29 @@ void registerMetricsForFullAuto(List<String> resources, HelixClusterManager heli
       registry.gauge(
           MetricRegistry.name(HelixClusterManager.class, "Resource_" + resource + "_TotalRegisteredHostDiskCapacity"),
           () -> resourceTotalRegisteredHostDiskCapacity);
+
+      Gauge<Integer> resourceTotalDiskCapacityUsage =
+          () -> helixClusterManager.getResourceTotalDiskCapacityUsage(resource);
+      registry.gauge(MetricRegistry.name(HelixClusterManager.class, "Resource_" + resource + "_TotalDiskCapacityUsage"),
+          () -> resourceTotalDiskCapacityUsage);
+
+      // register replica state metrics
+      for (ReplicaState replicaState : EnumSet.complementOf(EnumSet.of(ReplicaState.DROPPED))) {
+        Gauge<Integer> stateReplicaCountInResource =
+            () -> helixClusterManager.getReplicaCountForStateInResource(replicaState, resource);
+        registry.gauge(MetricRegistry.name(HelixClusterManager.class,
+            "Resource_" + resource + "_ReplicaCountInState" + replicaState.name()), () -> stateReplicaCountInResource);
```

Comment:
nit: may be additional `_` after `_ReplicaCountInState`?

Reply:
that's fine, the underscore is used for regular expression. The rest is following the camel pattern